lib/internal/streams/from.js

```diff
@@ -34,21 +34,52 @@ function from(Readable, iterable, opts) {
     objectMode: true,
     ...opts
   });
+
   // Reading boolean to protect against _read
   // being called before last iteration completion.
   let reading = false;
+
+  // needToClose boolean if iterator needs to be explicitly closed
+  let needToClose = false;
+
   readable._read = function() {
     if (!reading) {
       reading = true;
       next();
     }
   };
+
+  readable._destroy = async function(error, cb) {
+    try {
+      await close();
+    } catch (e) {
+      error = error || e;
+    } finally {
+      cb(error);
```

Comment:
I believe @mcollina's concern remains? The cb should probably be invoked through a nextTick?

Reply:
Hm. It seems so. Fixed.